### DISCUSSION TITLE
Auto-GPT-MetaTrader-Plugin

### DISCUSSION THREAD

**Discussion by isaiahbjork** [isaiahbjork/Auto-GPT-MetaTrader#5] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas about new features.
  * Share your strategies and experiences.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Con2802** [comment]
How well does it do?
How long does it hold?
Can i test it with a Demo Trading-Account?

**Reply by isaiahbjork** [reply]
I'm still testing it out so you will have to do the same and let us know what works for you. Always test on demo.

**Reply by Aadzo** [reply]
Has it been profitable so far ?

How long have you been testing it for ?

Does it learn with each trade ?

Which pairs have you tried it on ?

This is a very interesing project

**Comment by DarbDave** [comment]
Hello Isaiah, Questions...

- Does this work with MT4 & MT5?

- Do you know of any limitations with certain Brokers?

**Reply by isaiahbjork** [reply]
Yes works with both but seems to do better with MT4 and it's free to use. I haven't found any broker limitations yet.

**Reply by DarbDave** [reply]
Wow, cool, I actually gave the Link of your repository to Chat-GPT and asked it to tell me if it would work on MT4 and it said...
**"The Plugin I'm referring to is made by Isaiah Bjork and is located at:** 
https://github.com/isaiahbjork/Auto-GPT-MetaTrader-Plugin
Will this work on Metatrader 4?"

**It responded:**
"Based on the documentation on the GitHub page you provided, the Auto-GPT/MetaTrader Plugin is specifically designed for use with MetaTrader 5. Therefore, it is unlikely that it will work with MetaTrader 4. However, you can always contact the developer to confirm whether or not it is possible to make it work with MetaTrader 4."

**Comment by DarbDave** [comment]
I'm having a little problem understanding a couple of the API's
META_API_ACCOUNT_ID= This is the ID:  on the tile with my Broker Account, Correct?
META_API_TOKEN= Where do I find this?
META_API_REGION= Do I just type in New-York ?

**Reply by isaiahbjork** [reply]
> I'm having a little problem understanding a couple of the API's
> 
> META_API_ACCOUNT_ID= This is the ID:  on the tile with my Broker Account, Correct?
> 
> META_API_TOKEN= Where do I find this?
> 
> META_API_REGION= Do I just type in New-York ?

The account ID on MetaAPI and then the API key is under API Access. Yes "new-york".

**Comment by remixxan** [comment]
This is huge! Been looking for a way to integrate AutoGTP with trading software, such as MT4 and more. Even started to learn coding to make it happen somehow. So super cool to see you managed to get a plugin for it! 
I've got it installed and working, however, it seems that after a few cycles it hits its token limit and AutoGTP gives the message `

_> `"openai.error.InvalidRequestError: This model's maximum context length is 4097 tokens. However, your messages resulted in 4112 tokens. Please reduce the length of the messages."`

So I never get the result I'm looking for. Any idea to that Zay?
`_

**Reply by remixxan** [reply]
Yeah, it's not letting me continue any further. I go by the example:  

```
> Test Fetching Candlestick Data
Configure Auto-GPT: Set up Auto-GPT with the following parameters:

Name: TradeGPT
Role: fetch candlestick data
Goals:
Goal 1: fetch candlestick data for the 1 hour chart on EURUSD
Goal 2: Terminate
```
But I can't even finish that before the error message "maximum context length is 4097 tokens..." shows up and stops it. Am I doing it wrong or do you have a bypass to this Zay?

Edit,. I should add that I use` python -m autogpt --continuous --gpt3only ` if that helps

**Comment by rywager** [comment]
I'm getting "SYSTEM:  Command fetch_candlesticks returned: Failed to get candlesticks." 
From  COMMAND = fetch_candlesticks ARGUMENTS = {'symbol': 'EURUSD', 'timeframe': '1h'}

I'm not sure how I can see which of the API's are having issues, or what the error is.  Anyway to have this print a stack trace?

Would be kinda sick if it then used GPT to fix itself and recommended the setting change :)

Love this work, keep it up changing the world!

**Comment by hoffyroc** [comment]
File "plugins/Auto-GPT-MetaTrader-Plugin.zip/Auto-GPT-MetaTrader-Plugin/src/auto_gpt_metatrader/__init__.py", line 9, in <module>
ModuleNotFoundError: No module named 'ta' 


anyway to fix this?

**Reply by Silas38100** [reply]
Hi, I had the same problem, and it appeared that for me the TA file were located in a deep directory on my python file, I copied it from there and pasted it on root of autoGPT, no issue since!

**Reply by DarbDave** [reply]
Can you explain this a little more, I'm stuck with the Module Not Found (ta) error. Where is the "Deep Directory"

**Reply by DarbDave** [reply]
Well I'll be darned... I figured it out through search and it worked! Thanks

**Reply by mydaysx** [reply]
can u help to install it.. what autogpt version u use. with the latest 4.xxx i get class not found errors.
